### PR TITLE
Fix process hanging indefintely on Windows.

### LINKED ISSUE #8: Process APIs run indefinitely on Windows systems
Following Process APIs run indefinitely on windows systems:
- transformBytes 
- transformChunks
- thruExe
- thruExeChunks

### PR DESCRIPTION
**Pull request by pranaysashank**
Closes #8

### CODE REVIEW

File: src/Streamly/System/Process.hs [hunk 1]
```diff
@@ -85,13 +84,13 @@ openProc ::
     -> IO (Handle, ProcessHandle)
     -- ^ Handle to read from output of process, process handle
 openProc fpath args = do
-    (readEnd, writeEnd) <- createPipe
     let procObj = (proc fpath args) {
-            std_out = UseHandle writeEnd,
-            close_fds = True
+            std_out = CreatePipe,
```
Comment (pranaysashank) [inline src/Streamly/System/Process.hs:88]: The other handles i.e. `std_in` and `std_err` are set to inherit from the parent process, is this desirable?

Comment (harendra-kumar) [reply src/Streamly/System/Process.hs:88]: That sounds correct.

**Review by harendra-kumar — COMMENTED** [review]
When we use CreatePipe as stdout fd in CreateProcess, the windows pipe creation code sets inheritable on the pipe output:

```
  if (isInheritableOut) {
        if (!SetHandleInformation(hTemporaryOut,
                                  HANDLE_FLAG_INHERIT,
                                  HANDLE_FLAG_INHERIT))
        {
            maperrno();
            *pHandleIn  = NULL;
            *pHandleOut = NULL;
            CloseHandle(hTemporaryIn);
            CloseHandle(hTemporaryOut);
            return FALSE;
        }
    }
```

That may be the difference causing this issue?

File: test/Streamly/System/Process.hs [hunk 2]
```diff
@@ -85,21 +84,37 @@ arrayChunkElem :: Int
 arrayChunkElem = 100
 
 executableFile :: FilePath
+#if mingw32_HOST_OS == 1
```
[the remark this answers is not in the document]
Comment (pranaysashank) [reply test/Streamly/System/Process.hs:87]: yeah, it was failing due to a typo and I was just checking if I was doing it wrong.

**Pull request merged**